File: 1tdspm/aula53/ContatoControl.py
from Contato import Contato
from ContatoService import ContatoService

class ContatoControl:
    def __init__(self):
        self.service = ContatoService()
        self.lista = []
        self.ler_todos()

    def gravar(self, contato : Contato) -> bool:
        cadastrado = self.service.salvar(contato)
        self.ler_todos()
        return cadastrado

    def ler_todos(self) -> None:
        self.lista = self.service.ler_todos()

    def pesquisar_por_nome(self, parte_nome : str) -> Contato:
        for contato in self.lista:
            if parte_nome in contato.nome:
                return contato
        return None

File: 2tdspv/aula56/AgendaContato/app/src/main/java/edu/curso/agendacontato/ContatoControl.kt
package edu.curso.agendacontato

import android.app.Activity
import android.content.Context
import android.util.Log

class ContatoControl {

    val repository = ContatoRepository()
    val listaContatos = ArrayList<Contato>()

    fun salvar(contato : Contato, onSucesso : () -> Unit,
               onFalha : () -> Unit)  {
        repository.salvarFirebase( contato,
            sucesso = {
                lerTodos()
                onSucesso() },
            falha = { onFalha() } )

    }

    fun pesquisar( parteNome : String ) : Contato? {
        for (contato in listaContatos) {
            if(contato.nome.contains(parteNome)) {
                return contato
            }
        }
        return null
    }

    fun lerTodos() {
        repository.lerContatosFirebase(
            sucesso = {
                listaContatos.clear()
                listaContatos.addAll( it )
                Log.i("AGENDA", "ContatoControl recebeu lista com sucesso $listaContatos")
            },
            falha = {
                Log.e("AGENDA", "ContatoControl erro $it ao executar o LerTodos")
            })
    }
}